core/src/main/scala/com/rallyhealth/vapors/core/dsl/TimeFunctions.scala

```diff
@@ -0,0 +1,100 @@
+package com.rallyhealth.vapors.core.dsl
+
+import com.rallyhealth.vapors.core.algebra.{CaptureP, Expr}
+import com.rallyhealth.vapors.core.data.FactTable
+import com.rallyhealth.vapors.core.time.{CountTime, ModifyTime}
+import shapeless._
+
+import java.time.{Clock, Instant, LocalDate}
+
+trait TimeFunctions {
+
+  // TODO: Can this be converted to Addition[T] if we make the second argument a different type? Maybe AddLeft[T, D]?
+  final def dateAdd[V, T, D, P](
+    lhs: Expr[V, T, P],
+    rhs: Expr[V, D, P],
+  )(implicit
+    modTime: ModifyTime[T, D],
+    argTypeInfo: Typeable[T :: D :: HNil],
+    captureArgResult: CaptureP[V, T :: D :: HNil, P],
+    captureResult: CaptureP[V, T, P],
+  ): Expr[V, T, P] = {
+    val argExpr = wrap(lhs, rhs).asHList.returnOutput
+    val fn: T :: D :: HNil => T = {
+      case lhs :: rhs :: HNil =>
+        ModifyTime.addTo(lhs, rhs)
+    }
+    Expr.CustomFunction(argExpr, "date_add", fn, captureResult)
+  }
+
+  // TODO: Can this be converted to Addition[T] if we make the second argument a different type? Maybe AddLeft[T, D]?
```

Comment:
This is a pretty big change, I'm gonna make this change in a separate PR, targeted for 0.14.0.